### ISSUE TITLE
[jss-isolate] Regression in 4.0.1

### ISSUE THREAD

**Issue by sapegin** [cssinjs/jss#561]
Somehow selector isn’t updated and has only fraction of elements.

4.0.0:

```
.root-0-1,
.hasSidebar-0-2,
.content-0-3,
.sidebar-0-4,
.logo-0-5,
.footer-0-6,
.root-0-8,
.root-0-9,
.root-0-10,
.header-0-11,
.tabs-0-12,
.tabButtons-0-13,
.docs-0-14,
.root-0-15,
.heading-0-16,
.isPrimary-0-17,
.isDeprecated-0-18,
.toolbar-0-19,
.button-0-20,
.button-0-20 svg,
.isSmall-0-21,
.isSmall-0-21 svg,
.pathline-0-22,
.copyButton-0-23,
.base-0-24,
.para-0-25,
.a-0-26,
.a-0-26, .a-0-26:link, .a-0-26:visited,
.h1-0-27,
.h2-0-28,
.h3-0-29,
.h4-0-30,
.h5-0-31,
.h6-0-32,
.p-0-33,
.ul-0-34,
.ol-0-35,
.li-0-36,
.input-0-37,
.blockquote-0-38,
.hr-0-39,
.em-0-40,
.strong-0-41,
.code-0-42,
.pre-0-43,
.table-0-44,
.thead-0-45,
.tbody-0-46,
.td-0-47,
.th-0-48,
.tr-0-49,
.button-0-50,
.isActive-0-51,
.root-0-52,
.root-0-53,
.preview-0-54,
.controls-0-55,
.toolbar-0-56,
.para-0-57,
.logo-0-58,
.root-0-59,
.search-0-60,
.input-0-61,
.list-0-62,
.item-0-63,
.isChild-0-64,
.heading-0-65,
.link-0-66,
.link-0-66, .link-0-66:link, .link-0-66:visited
```

4.0.1:

```
.root-0-1,
.hasSidebar-0-2,
.content-0-3,
.sidebar-0-4,
.logo-0-5,
.footer-0-6
```

You can reproduce it in [react-styleguidist](https://github.com/styleguidist/react-styleguidist) repo like this:

```
rm package-lock.json
npm i
npm start
```

I’ve tried to debug it but couldn’t find the reason. `setSelector` is called with the right list of elements but the attached stylesheet isn’t updated.

P. S. What’s `debaunceMaybe` mentioned in the comments?

**Comment by kof** [comment]
which jss version was used?

**Comment by sapegin** [comment]
8.1.0

**Comment by kof** [comment]
also, did you look in elements view? if yes look into style view of some element with a class

**Comment by sapegin** [comment]
That’s where I’ve copied the selector from ;-) Actually here it is deployed: https://react-styleguidist.js.org/examples/basic/

**Comment by kof** [comment]
will look into it soon

**Comment by sapegin** [comment]
![image 2017-08-11 at 10 39 08 am](https://user-images.githubusercontent.com/70067/29206241-531cf1ce-7e81-11e7-8847-3545ad51ec79.png)

**Comment by kof** [comment]
Just debugged it. As the timing has changed when isolated classes are added to a reset rule, you might see a shorter list when the rule to isolate is not actually attached. Also don't look into "elements tab" in dom inspector, look into the styles applied to a particular element. Depending on when it is added, it might be not visible through the elements tab because this one is showing you the textContent representation of  a style element, while rules added using insertRule api are not showed there, but you can see them in the styles applied to an element.


We might solve the debugging at some point, I haven't found a proper way so far though https://github.com/cssinjs/jss/issues/491

**Comment by kof** [comment]
Please reopen if you think there is still an issue.

**Comment by sapegin** [comment]
I woudn’t look into elements tab if the behavior was correct ;-) It was working fine in 4.0.0 but not in 4.0.1.

Please try:

```
git clone git@github.com:styleguidist/react-styleguidist.git
cd react-styleguidist
npm i
npm i jss-isolate@4.0.1
npm start
```

And compare with the correct rendering [here](https://react-styleguidist.js.org/examples/basic/), especially the search box and copy path icon.

**Comment by kof** [comment]
found the visual difference now, looking into it again

**Comment by kof** [comment]
Ok, found a bug, it has worked for me because I am using jss 9 pre release.

**Comment by kof** [comment]
Now I either need to fix it or to republish jss-isolate 4.0.2 with a reverted fix and publish jss-usolate 5 pre which will be using jss 9 pre.

**Comment by kof** [comment]
ok, releasing 4.0.2 which reverts the changes of 4.0.1

**Comment by kof** [comment]
released jss-isolate@5.0.0-pre.1
